app/src/androidTest/java/ch/hikemate/app/endtoend/EndToEndTest.kt

```diff
@@ -1,40 +1,77 @@
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
 import ch.hikemate.app.MainActivity
 import ch.hikemate.app.ui.auth.SignInScreen
+import ch.hikemate.app.ui.auth.SignInWithEmailScreen
 import ch.hikemate.app.ui.map.MapScreen.TEST_TAG_MAP
 import ch.hikemate.app.ui.navigation.LIST_TOP_LEVEL_DESTINATIONS
+import ch.hikemate.app.ui.navigation.Screen
 import ch.hikemate.app.ui.navigation.Screen.PROFILE
 import ch.hikemate.app.ui.navigation.TEST_TAG_DRAWER_CLOSE_BUTTON
 import ch.hikemate.app.ui.navigation.TEST_TAG_DRAWER_CONTENT
 import ch.hikemate.app.ui.navigation.TEST_TAG_DRAWER_ITEM_PREFIX
 import ch.hikemate.app.ui.navigation.TEST_TAG_SIDEBAR_BUTTON
 import ch.hikemate.app.ui.navigation.TopLevelDestinations
 import ch.hikemate.app.ui.saved.TEST_TAG_SAVED_HIKES_SECTION_CONTAINER
+import com.google.firebase.FirebaseApp
+import com.google.firebase.auth.FirebaseAuth
+import junit.framework.TestCase
+import org.junit.After
+import org.junit.Before
 import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
 
-// @RunWith(AndroidJUnit4::class)
-// TODO: Change it to "EndToEndTest: TestCase()"
-class EndToEndTest {
+@RunWith(AndroidJUnit4::class)
+class EndToEndTest : TestCase() {
   @get:Rule val composeTestRule = createAndroidComposeRule<MainActivity>()
+  private val auth = FirebaseAuth.getInstance()
+  private val email = "end2endTest@gmail.com"
+  private val password = "password"
 
-  // @Test
+  @Before
+  fun setupFirebase() {
+    val context = ApplicationProvider.getApplicationContext<Context>()
+    FirebaseApp.initializeApp(context)
+
+    auth.createUserWithEmailAndPassword(email, password)
+    auth.signOut()
+  }
+
+  @After
+  fun deleteUser() {
+    auth.currentUser?.delete()
+    auth.signOut()
+  }
+
+  @Test
   fun test() {
     // Check that we are on the login screen
     composeTestRule.onNodeWithTag(SignInScreen.TEST_TAG_SIGN_IN_WITH_EMAIL).assertIsDisplayed()
     composeTestRule.onNodeWithTag(SignInScreen.TEST_TAG_SIGN_IN_WITH_GOOGLE).assertIsDisplayed()
 
-    // Click on the login button
-    composeTestRule.onNodeWithTag(SignInScreen.TEST_TAG_SIGN_IN_WITH_GOOGLE).performClick()
+    composeTestRule.onNodeWithTag(SignInScreen.TEST_TAG_SIGN_IN_WITH_EMAIL).performClick()
+    composeTestRule.onNodeWithTag(Screen.SIGN_IN_WITH_EMAIL).assertIsDisplayed()
+    composeTestRule
+        .onNodeWithTag(SignInWithEmailScreen.TEST_TAG_EMAIL_INPUT)
+        .performTextInput(email)
+    composeTestRule
+        .onNodeWithTag(SignInWithEmailScreen.TEST_TAG_PASSWORD_INPUT)
+        .performTextInput(password)
+    composeTestRule.onNodeWithTag(SignInWithEmailScreen.TEST_TAG_SIGN_IN_BUTTON).performClick()
 
-    // Check that we are not on the login screen anymore
-    composeTestRule.onNodeWithTag(SignInScreen.TEST_TAG_SIGN_IN_WITH_GOOGLE).assertIsNotDisplayed()
+    // Wait for the map to load
+    Thread.sleep(1000)
```

Comment:
Question: Have you tried if `composeTestRule.waitForIdle()` isn't enough? It would be better that a `Thread.sleep` if it works

Reply:
I just tried and this doesn't seems to work. I think 1 second delay is fair for this kind of test (we won't have much of them)

Reply:
Okay, thanks for the try!